File: 01_big_O_notation/02_quadratic_cost/quadratic_cost.js
function quadraticCost(n) {
	for (let i = 0; i < n; i++) {
		console.log(i)

		for (let j = 0; j < n; j++) {
			console.log(j)
		}
	}
}

function quadraticCostWay02(n) {
	for (let i = 0; i < n ** 2; i++) {
		console.log(i)
	}
}

quadraticCost(3)
quadraticCostWay02(3)

File: 01_big_O_notation/02_quadratic_cost/quadratic_cost.c
// O(n²)

#include <math.h>
#include <stdio.h>
#include <stdlib.h>

void quadratic_cost(int n) {
  for (int i = 0; i < n; i++) {
    printf("\n%d\n", i);

    for (int j = 0; j < n; j++) {
      printf("%d\n", j);
    }
  }
}

void quadratic_cost_way_02(int n) {
  for (int i = 0; i < pow(n, 2); i++) {
    printf("%d\n", i);
  }
}

int main(void) {
  // quadratic_cost(3);
  quadratic_cost_way_02(3);

  return 0;
}